### ISSUE TITLE
How to get the tensor‘s shape, just like .shape() in numpy?

### ISSUE THREAD

**Issue by meanmee** [fchollet/keras#2095]
I have tried tf.shape(), but it returned sth like:  Tensor("Shape:0", shape=(2,), dtype=int32)
I want to define a custom loss function, which need the batch size(y_pred.shape[0])

**Comment by ymcui** [comment]
`Layer` object has `output_shape()` function, it will return the output shape of the layer.
for e.g.,
`model.layers[-1].output_shape()` will return the final output shape of `model`

**Comment by tboquet** [comment]
You could find information [here](https://github.com/fchollet/keras/blob/master/keras/backend/tensorflow_backend.py#L43-L50) and [here](https://github.com/fchollet/keras/blob/master/keras/backend/theano_backend.py#L36-L42).
